core/include/scipp/core/variable.h

```diff
@@ -1054,6 +1056,11 @@ namespace geometry {
 [[nodiscard]] SCIPP_CORE_EXPORT Variable x(const VariableConstView &pos);
 [[nodiscard]] SCIPP_CORE_EXPORT Variable y(const VariableConstView &pos);
 [[nodiscard]] SCIPP_CORE_EXPORT Variable z(const VariableConstView &pos);
+[[nodiscard]] SCIPP_CORE_EXPORT Variable rotate(const VariableConstView &pos,
+                                                const VariableConstView &rot);
+SCIPP_CORE_EXPORT VariableView rotate(const VariableConstView &pos,
```

Comment:
Maybe for another PR, but would it be nice to unpack/pack the quaternion to/from to a more naturally human readable representation maybe utilising `Eigen::AngleAxis` which can take a `Quaternion` a constructor argument. This would give functionality much like existing `geometry.position` and `geometry::x,...` which could all be done with `transform`

Reply:
I guess this would mean having to also add python bindings for AngleAxis

Reply:
I imagined that `Eigen::AxisAngle` would be only be needed internal, i.e a 4 variables defining rotation angle + unit vector (3) could be zipped to a single quaternion, the reverse transform(s) could also be provided. That seems convenient, but since we don't yet have any need for it, I think it's somewhat academic.